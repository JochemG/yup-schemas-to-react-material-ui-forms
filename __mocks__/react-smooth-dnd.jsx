const React = require('react');

module.exports.Container = function Container({children, onDrop}) {
    testOnDrops.push(onDrop);
    return React.createElement('div',{children});
}
module.exports.Draggable = function Draggable({children}) {
    return React.createElement('div',{children});
}
const testOnDrops = [];
module.exports.testOnDrops = testOnDrops;