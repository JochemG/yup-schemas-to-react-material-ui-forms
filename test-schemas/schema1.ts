import {object, string, number, date, array, bool} from 'yup';

export const schema = object().shape({
    user: object({
        firstName: string().required(),
        age: number().required(),
        dob: date().required(),
        animal: array().of(object().shape({
            name: string().required(),
            legs: number().required()
        })),
        hasGlasses: bool().required()
    })
})