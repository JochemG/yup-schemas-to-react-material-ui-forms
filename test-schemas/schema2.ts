import {object, string, number, date, array, bool} from 'yup';

export const schema = object().shape({
    arrayOfObjects: array().of(object({something: string(), somethingElse: string()})),
    arrayOfString: array().of(string()),
    arrayOfBooleans: array().of(bool()),
    arrayOfNumber: array().of(number()),
    arrayOfDate: array().of(date()),
    arrayOfArrayOfString: array().of(array().of(string())),
})