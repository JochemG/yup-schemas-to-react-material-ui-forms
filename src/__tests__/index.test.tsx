import React from "react";
import {AnySchema, BaseSchema} from "yup"
import {FormFromSchema} from '../index'
import {schema as schema1} from '../../test-schemas/schema1';
import {render, act as reactAct, fireEvent, getByText} from '@testing-library/react';
// @ts-ignore
import {testOnDrops} from "react-smooth-dnd";
jest
    .useFakeTimers('modern')
    .setSystemTime(new Date('2020-01-01').getTime());

describe('FormFromSchema', () => {
    beforeEach(() => {
        testOnDrops.splice(0);
    })
    type ActProperties = {
        schema: AnySchema,
        value: any,
        actions?: ({container: HTMLElement}) => void
    }
    function act({schema, value, actions}: ActProperties) {
        const onSubmit = jest.fn();
        const rendered = render(<FormFromSchema schema={schema} value={value} onSubmit={onSubmit}/>)
        const {asFragment, container} = rendered;
        actions && actions({container});
        fireEvent.click(getByText(container,"Save"))
        return {
            toAssert: () => ({
                asFragment: asFragment(),
                onSubmitCalls: onSubmit.mock.calls.map(call => [call[0], call[1].nativeEvent.toString(), call.length])
            })
        }
    }

    describe('schema1', () => {
        test(`Render with values`, () => {
            expect.assertions(1);
            const actions = ({container}: {container: HTMLElement}) => {
                // Add cat with 4 legs to end of animals list
                fireEvent.click(container.getElementsByClassName("addListItem")[0]);
                fireEvent.change(container.querySelector('input[name="user.animal.2.name"]'), { target: { value: 'Kitty cat' } });
                fireEvent.change(container.querySelector('input[name="user.animal.2.legs"]'), { target: { value: '4' } });
                // Delete the dog (first item)
                fireEvent.click(container.getElementsByClassName("deleteListItem")[0]);
                // Swap cat & bird
                reactAct(() => {testOnDrops[0]({removedIndex: 1, addedIndex: 0})});
            };
            const {toAssert} = act({
                schema: schema1,
                value: {
                    user: {
                        firstName: 'John',
                        age: 28,
                        dob: new Date(0),
                        animal: [{
                            name: 'Jack doggo',
                            legs: 4
                        }, {
                            name: 'Coco birdie',
                            legs: 2
                        }],
                        hasGlasses: true,
                        country: "Belgium"
                    }
                },
                actions
            });

            expect(toAssert()).toMatchSnapshot();
        });
        test(`Render without values`, () => {
            expect.assertions(1);
            const {toAssert} = act({
                schema: schema1,
                value: undefined
            });

            expect(toAssert()).toMatchSnapshot();
        });
        test(`Render with user == null`, () => {
            expect.assertions(1);
            const {toAssert} = act({
                schema: schema1,
                value: {user: null}
            });

            expect(toAssert()).toMatchSnapshot();
        });
    })

    describe('invalid schema item', () => {
        class SomeNewYupType extends BaseSchema {
            constructor() {
                super({type: "someNewYupType"});
            }
        }
        function someNewYupType() {
            return new SomeNewYupType();
        }
        test('Render', () => {
            expect.assertions(1);

            const {toAssert} = act({
                schema: someNewYupType(),
                value: undefined
            });

            expect(toAssert()).toMatchSnapshot();
        });
    });
})
export {};