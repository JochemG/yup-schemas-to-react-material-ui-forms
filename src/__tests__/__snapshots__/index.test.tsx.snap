// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFromSchema invalid schema item Render 1`] = `
Object {
  "asFragment": <DocumentFragment>
    <form>
      <div>
        Type not supported. name: , schema.type: someNewYupType
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Save
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </DocumentFragment>,
  "onSubmitCalls": Array [
    Array [
      undefined,
      "[object Event]",
      2,
    ],
  ],
}
`;

exports[`FormFromSchema schema1 Render with user == null 1`] = `
Object {
  "asFragment": <DocumentFragment>
    <form>
      <ul
        class="MuiList-root fullWidth MuiList-padding"
      >
        <div
          class="objectChildren"
        >
          <ul
            class="MuiList-root fullWidth MuiList-padding MuiList-subheader"
          >
            <li
              class="MuiListSubheader-root"
            >
              user
            </li>
            <div
              class="objectChildren"
            >
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    firstName
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="user.firstName"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    age
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="user.age"
                      type="number"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    dob
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                      name="user.dob"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <ul
                class="MuiList-root fullWidth MuiList-padding MuiList-subheader"
              >
                <li
                  class="MuiListSubheader-root"
                >
                  animal
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root addIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                    />
                  </svg>
                </li>
                <div />
              </ul>
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-12"
                      data-indeterminate="false"
                      name="user.hasGlasses"
                      type="checkbox"
                      value="true"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  hasGlasses
                </span>
              </label>
            </div>
          </ul>
        </div>
      </ul>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Save
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </DocumentFragment>,
  "onSubmitCalls": Array [
    Array [
      Object {
        "user": Object {
          "age": 0,
          "animal": Array [],
          "dob": "",
          "firstName": "",
          "hasGlasses": false,
        },
      },
      "[object Event]",
      2,
    ],
  ],
}
`;

exports[`FormFromSchema schema1 Render with values 1`] = `
Object {
  "asFragment": <DocumentFragment>
    <form>
      <ul
        class="MuiList-root fullWidth MuiList-padding"
      >
        <div
          class="objectChildren"
        >
          <ul
            class="MuiList-root fullWidth MuiList-padding MuiList-subheader"
          >
            <li
              class="MuiListSubheader-root"
            >
              user
            </li>
            <div
              class="objectChildren"
            >
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    firstName
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="user.firstName"
                      type="text"
                      value="John"
                    />
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    age
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="user.age"
                      type="number"
                      value="28"
                    />
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    dob
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                      name="user.dob"
                      type="text"
                      value="01/01/1970"
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <ul
                class="MuiList-root fullWidth MuiList-padding MuiList-subheader"
              >
                <li
                  class="MuiListSubheader-root"
                >
                  animal
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root addIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                    />
                  </svg>
                </li>
                <div>
                  <div>
                    <li
                      class="MuiListItem-container"
                    >
                      <div
                        class="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
                      >
                        <ul
                          class="MuiList-root fullWidth MuiList-padding MuiList-subheader"
                        >
                          <li
                            class="MuiListSubheader-root"
                          >
                            0
                          </li>
                          <div
                            class="objectChildren"
                          >
                            <div>
                              <div
                                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                              >
                                <label
                                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                                  data-shrink="true"
                                >
                                  name
                                </label>
                                <div
                                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                                >
                                  <input
                                    aria-invalid="false"
                                    class="MuiInputBase-input MuiInput-input"
                                    name="user.animal.0.name"
                                    type="text"
                                    value=""
                                  />
                                </div>
                              </div>
                            </div>
                            <div>
                              <div
                                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                              >
                                <label
                                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                                  data-shrink="true"
                                >
                                  legs
                                </label>
                                <div
                                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                                >
                                  <input
                                    aria-invalid="false"
                                    class="MuiInputBase-input MuiInput-input"
                                    name="user.animal.0.legs"
                                    type="number"
                                    value=""
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                        </ul>
                      </div>
                      <div
                        class="MuiListItemSecondaryAction-root"
                      >
                        <div
                          class="MuiListItemIcon-root drag-handle noIconSpacing"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M20 9H4v2h16V9zM4 15h16v-2H4v2z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiListItemIcon-root noIconSpacing"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root deleteListItem"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                            />
                          </svg>
                        </div>
                      </div>
                    </li>
                  </div>
                  <div>
                    <li
                      class="MuiListItem-container"
                    >
                      <div
                        class="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
                      >
                        <ul
                          class="MuiList-root fullWidth MuiList-padding MuiList-subheader"
                        >
                          <li
                            class="MuiListSubheader-root"
                          >
                            1
                          </li>
                          <div
                            class="objectChildren"
                          >
                            <div>
                              <div
                                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                              >
                                <label
                                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                                  data-shrink="true"
                                >
                                  name
                                </label>
                                <div
                                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                                >
                                  <input
                                    aria-invalid="false"
                                    class="MuiInputBase-input MuiInput-input"
                                    name="user.animal.1.name"
                                    type="text"
                                    value="Coco birdie"
                                  />
                                </div>
                              </div>
                            </div>
                            <div>
                              <div
                                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                              >
                                <label
                                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                                  data-shrink="true"
                                >
                                  legs
                                </label>
                                <div
                                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                                >
                                  <input
                                    aria-invalid="false"
                                    class="MuiInputBase-input MuiInput-input"
                                    name="user.animal.1.legs"
                                    type="number"
                                    value="2"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                        </ul>
                      </div>
                      <div
                        class="MuiListItemSecondaryAction-root"
                      >
                        <div
                          class="MuiListItemIcon-root drag-handle noIconSpacing"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M20 9H4v2h16V9zM4 15h16v-2H4v2z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiListItemIcon-root noIconSpacing"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root deleteListItem"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                            />
                          </svg>
                        </div>
                      </div>
                    </li>
                  </div>
                </div>
              </ul>
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input-4"
                      data-indeterminate="false"
                      name="user.hasGlasses"
                      type="checkbox"
                      value="true"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  hasGlasses
                </span>
              </label>
            </div>
          </ul>
        </div>
      </ul>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Save
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </DocumentFragment>,
  "onSubmitCalls": Array [
    Array [
      Object {
        "user": Object {
          "age": 28,
          "animal": Array [
            Object {
              "legs": 4,
              "name": "Kitty cat",
            },
            Object {
              "legs": 2,
              "name": "Coco birdie",
            },
          ],
          "dob": "01/01/1970",
          "firstName": "John",
          "hasGlasses": true,
        },
      },
      "[object Event]",
      2,
    ],
  ],
}
`;

exports[`FormFromSchema schema1 Render without values 1`] = `
Object {
  "asFragment": <DocumentFragment>
    <form>
      <ul
        class="MuiList-root fullWidth MuiList-padding"
      >
        <div
          class="objectChildren"
        >
          <ul
            class="MuiList-root fullWidth MuiList-padding MuiList-subheader"
          >
            <li
              class="MuiListSubheader-root"
            >
              user
            </li>
            <div
              class="objectChildren"
            >
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    firstName
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="user.firstName"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    age
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="user.age"
                      type="number"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    dob
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                      name="user.dob"
                      type="text"
                      value="01/01/2020"
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <ul
                class="MuiList-root fullWidth MuiList-padding MuiList-subheader"
              >
                <li
                  class="MuiListSubheader-root"
                >
                  animal
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root addIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                    />
                  </svg>
                </li>
                <div />
              </ul>
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-8"
                      data-indeterminate="false"
                      name="user.hasGlasses"
                      type="checkbox"
                      value="true"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  hasGlasses
                </span>
              </label>
            </div>
          </ul>
        </div>
      </ul>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Save
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </DocumentFragment>,
  "onSubmitCalls": Array [
    Array [
      Object {
        "user": Object {
          "age": 0,
          "animal": Array [],
          "dob": "01/01/2020",
          "firstName": "",
          "hasGlasses": false,
        },
      },
      "[object Event]",
      2,
    ],
  ],
}
`;
